### PR TITLE
13698 - Do all the nitty gritty with postprocessing and Unity's xcodeproj file.

### PR DESCRIPTION
**Pull request by seaders**

### CODE REVIEW

File: unity3d/Assets/Swrve/Editor/SwrvePostProcess.cs [hunk 1]
```diff
@@ -45,10 +50,57 @@ public static void OnPostprocessBuild(BuildTarget target, string pathToBuiltProj
...
+
+        Match match = Regex.Match (project, pattern, RegexOptions.Multiline);
+        log (string.Format ("searching for {0}, found: {1}", pattern, match.Success));
+
+        project = Regex.Replace (project, pattern, replacement, RegexOptions.Multiline);
+
+        return project;
+    }
+
+    private static string FixList (string grouping, string project)
```
Comment (Sergio-Mira) [inline unity3d/Assets/Swrve/Editor/SwrvePostProcess.cs]:
~~~ Rename to something like AddInheritedToList?

File: unity3d/Assets/Swrve/Editor/SwrvePostProcess.cs [hunk 2]
```diff
@@ -45,10 +50,57 @@ public static void OnPostprocessBuild(BuildTarget target, string pathToBuiltProj
...
+    private static void CorrectXCodeProject (string pathToProject, bool writeOut)
+    {
+        string path = Path.Combine (Path.Combine (pathToProject, "Unity-iPhone.xcodeproj"), "project.pbxproj");
+        string xcodeproj = File.ReadAllText (path);
+
+        xcodeproj = FixList ("OTHER_CFLAGS", xcodeproj);
+        xcodeproj = FixList ("OTHER_LDFLAGS", xcodeproj);
+        xcodeproj = FixList ("HEADER_SEARCH_PATHS", xcodeproj);
+        xcodeproj = FixList ("GCC_PREPROCESSOR_DEFINITIONS", xcodeproj);
+        xcodeproj = FixValue ("SDKROOT", xcodeproj, "\"iphoneos\"");
```
Comment (Sergio-Mira) [inline unity3d/Assets/Swrve/Editor/SwrvePostProcess.cs]: Will this interfere when building for the simulator?

**Comment by Sergio-Mira** [comment]
🚀  (after resolving those 2 comments)

File: unity3d/Assets/Swrve/Editor/SwrvePostProcess.cs (hunk 2 above)
Comment (seaders) [inline unity3d/Assets/Swrve/Editor/SwrvePostProcess.cs]: It _specifically_ fixes for the simulator.  When Unity builds for there, it sets SDKROOT to "iphonesimulator" which XCode doesn't recognise, and doesn't have a clue how to build for!

**Pull request merged**